Lib/test/test_abc.py

```diff
@@ -326,7 +326,7 @@ class B:
             token_old = abc_get_cache_token()
             A.register(B)
             token_new = abc_get_cache_token()
-            self.assertNotEqual(token_old, token_new)
+            self.assertGreater(token_new, token_old)
```

Comment:
This test might be what we wanted exactly.
I just updated the test for more validation.